### PR TITLE
Fix Leech Seed on Allies

### PR DESCRIPTION
**Pull request by ascriptmaster**
When Leech Seed targets an ally in Double Battles, it should heal the
Pokemon at the ally's position, not an enemy.

### CODE REVIEW

**Comment by ascriptmaster** [comment]
Hmm, I'm not actually sure if the eventData is set by the time the Start event is called. I'll double check after lunch.

File: data/moves.js [hunk 1]
```diff
@@ -7709,10 +7709,11 @@ exports.BattleMovedex = {
 		effect: {
 			onStart: function (target) {
 				this.add('-start', target, 'move: Leech Seed');
+				this.effectData.sourceSide = this.effectData.source.side;
 			},
 			onResidualOrder: 8,
 			onResidual: function (pokemon) {
-				var target = pokemon.side.foe.active[pokemon.volatiles['leechseed'].sourcePosition];
+				var target = this.effectData.sourceSide.active[pokemon.volatiles['leechseed'].sourcePosition];
```
Comment (Zarel) [inline data/moves.js]: Why are you caching `sourceSide`? Can't you just do `.source.side` each time here?

**Comment by Zarel** [comment]
Wait what? Haven't you tested this? o_O

**Comment by ascriptmaster** [comment]
lol, you got me. PS is blocked where I'm at right now and I'm too lazy to VPN

I'll get back to this in the evening, mayyyybe.

**Comment by ascriptmaster** [comment]
Re-opening because I have the fix now.

Commit incoming.

**Comment by ascriptmaster** [comment]
wait, shit, dirtied branch

**Pull request closed without merging**